pkg/cloud/utils.go

```diff
@@ -93,29 +99,47 @@ func (r *ResourceID) MapKey() ResourceMapKey {
 
 // RelativeResourceName returns the relative resource name string
 // representing this ResourceID.
+// Deprecated: Use SelfLinkWithGroup instead
 func (r *ResourceID) RelativeResourceName() string {
 	return RelativeResourceName(r.ProjectID, r.Resource, r.Key)
 }
 
 // ResourcePath returns the resource path representing this ResourceID.
+// Deprecated: Use SelfLinkWithGroup instead
 func (r *ResourceID) ResourcePath() string {
 	return ResourcePath(r.Resource, r.Key)
 }
 
+// SelfLink returns a URL representing the resource and assumes Compute API Group.
+// Deprecated: Use SelfLinkWithGroup instead
```

Comment:
This is pretty confusing to me. Why do we need to change the contract for a ResourceID? The ResourceID . SelfLink() is already a fully qualified URL including the API Group.

Why can't we have this method treat an empty API group as compute API?

I think we need to discuss this change as this approach feels wrong.

Reply:
Yep, that's fair, I was trying to mirror the other functions but you're right that it's unnecessary here. I've reworked this and added new tests.